src/scripts/main.js

```diff
@@ -1,3 +1,65 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('.field');
+const tbody = table.querySelector('tbody');
+const appendRow = document.querySelector('.append-row');
+const removeRow = document.querySelector('.remove-row');
+const appendColumn = document.querySelector('.append-column');
+const removeColumn = document.querySelector('.remove-column');
+
+function updateButtonStates() {
+  const trs = tbody.querySelectorAll('tr');
+  const rowCount = trs.length;
+  const columnCount = trs[0].children.length;
```

Comment:
Ensure that `trs` is not empty before accessing `trs[0]`. If the table is initially empty, this line will throw an error. Consider adding a check to handle cases where `trs.length` is 0.